File: js/monde/entites/Dechet.js
(function(){

if(!window.creatura)
	window.creatura={};

/**
 * Représente un déchet (un excrément en fait) dans le monde (étend creatura.Entite)
 */
creatura.Dechet = function(monde, quantite, x, y){
	creatura.Entite.call(this, monde, x, y, quantite/1000.0, quantite/1000.0);
	this.quantite = quantite;
	
	/**
	 * Renvoie le type de l'entité (TODO : Enum ?)
	 */
	this.getType = function(){
		return "dechet";
	};

	/**
	 * Renvoie la quantité de déchets pouvant être contenus dans cette litiere
	 */
	this.getTaille = function(){
		return this.quantite;
	};

	/**
	 * Met à jour l'entité
	 * @param frame : la frame de mise à jour
	 */
	this.metAJour = function(frame){}; //Inerte.
};

})();

File: src/net/feerie/creatura/shared/entites/Dechet.java
package net.feerie.creatura.shared.entites;

import net.feerie.creatura.shared.Monde;
import net.feerie.creatura.shared.commons.Dimension;
import net.feerie.creatura.shared.commons.Position;

/**
 * Représente un déchet (une crotte)
 * 
 * @author greewi
 */
public class Dechet extends Entite
{
	
	/**
	 * @param monde le monde de cette entité
	 * @param position la position de cette entité
	 */
	public Dechet(Monde monde, Position position)
	{
		super(monde, position, new Dimension(2, 2));
	}
	
	@Override
	public TypeEntite getType()
	{
		return TypeEntite.POPO;
	}
	
	@Override
	public void metAJour(int frame)
	{
		//Inerte
	}
	
}
